# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/compiler/translator/glslang_lex.cpp
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/compiler/translator/glslang_lex.cpp
--- a/gfx/angle/checkout/src/compiler/translator/glslang_lex.cpp	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/compiler/translator/glslang_lex.cpp	Fri Nov 08 17:06:00 2019 +0000
@@ -1221,10 +1221,10 @@
 static int ES2_reserved_ES3_keyword(TParseContext *context, int token);
 static int ES2_keyword_ES3_reserved(TParseContext *context, int token);
 static int ES2_ident_ES3_keyword(TParseContext *context, int token);
-static int ES2_ident_ES3_keyword_multiview_keyword(TParseContext *context, int token);
 static int ES2_ident_ES3_reserved_ES3_1_keyword(TParseContext *context, int token);
 static int ES2_and_ES3_reserved_ES3_1_keyword(TParseContext *context, int token);
 static int ES2_and_ES3_ident_ES3_1_keyword(TParseContext *context, int token);
+static int ES2_extension_ES3_keyword_else_reserved(TParseContext *context, TExtension extension, int token);
 static int ES3_extension_keyword_else_ident(TParseContext *context, TExtension extension, int token);
 static int ES2_ident_ES3_reserved_ES3_1_extension_keyword(TParseContext *context, TExtension extension, int token);
 static int ES3_extension_and_ES3_1_keyword_ES3_reserved_else_ident(TParseContext *context, TExtension extension, int token);
@@ -1955,7 +1955,7 @@
 	YY_BREAK
 case 64:
 YY_RULE_SETUP
-{ return ES2_reserved_ES3_keyword(context, SAMPLER3D); }
+{ return ES2_extension_ES3_keyword_else_reserved(context, TExtension::OES_texture_3D, SAMPLER3D); }
 	YY_BREAK
 case 65:
 YY_RULE_SETUP
@@ -2047,7 +2047,7 @@
 	YY_BREAK
 case 87:
 YY_RULE_SETUP
-{ return ES2_ident_ES3_keyword_multiview_keyword(context, LAYOUT); }
+{ return ES2_ident_ES3_keyword(context, LAYOUT); }
 	YY_BREAK
 case 88:
 YY_RULE_SETUP
@@ -3888,22 +3888,6 @@
     return token;
 }
 
-int ES2_ident_ES3_keyword_multiview_keyword(TParseContext *context, int token)
-{
-    struct yyguts_t* yyg = (struct yyguts_t*) context->getScanner();
-    yyscan_t yyscanner = (yyscan_t) context->getScanner();
-
-    // not a reserved word in GLSL ES 1.00, so could be used as an identifier/type name
-    // except when multiview extension is enabled
-    if (context->getShaderVersion() < 300 && !context->isExtensionEnabled(TExtension::OVR_multiview2))
-    {
-        yylval->lex.string = AllocatePoolCharArray(yytext, yyleng);
-        return check_type(yyscanner);
-    }
-
-    return token;
-}
-
 int ES2_and_ES3_reserved_ES3_1_keyword(TParseContext *context, int token)
 {
     yyscan_t yyscanner = (yyscan_t) context->getScanner();
@@ -3931,6 +3915,19 @@
     return token;
 }
 
+int ES2_extension_ES3_keyword_else_reserved(TParseContext *context, TExtension extension, int token)
+{
+    yyscan_t yyscanner = (yyscan_t) context->getScanner();
+
+    // Available with extension or ES 3.00 and above, reserved otherwise
+    if (context->isExtensionEnabled(extension) || context->getShaderVersion() >= 300)
+    {
+        return token;
+    }
+
+    return reserved_word(yyscanner);
+}
+
 int ES3_extension_keyword_else_ident(TParseContext *context, TExtension extension, int token)
 {
     struct yyguts_t* yyg = (struct yyguts_t*) context->getScanner();
@@ -4109,4 +4106,3 @@
     return 0;
 }
 
- 
\ No newline at end of file